extension/apikeyauthextension/authenticator.go

```diff
@@ -225,12 +227,12 @@ func (a *authenticator) getCacheKey(id string, headers map[string][]string) (str
 func (a *authenticator) Authenticate(ctx context.Context, headers map[string][]string) (context.Context, error) {
 	authHeaderValue, id, err := a.parseAuthorizationHeader(headers)
 	if err != nil {
-		return ctx, err
+		return ctx, status.Error(codes.Unauthenticated, err.Error())
 	}
 
 	cacheKey, err := a.getCacheKey(id, headers)
 	if err != nil {
-		return ctx, err
+		return ctx, status.Error(codes.Internal, err.Error())
```

Comment:
Doesn't really hurt, but I'd be inclined to not use `status.Error` here or in the other Internal case, and expect callers to assume errors are internal unless told otherwise

Reply:
good point